src/rpc/mining.cpp

```diff
@@ -548,11 +550,17 @@ UniValue getblocktemplate(const UniValue& params, bool fHelp)
     if (IsInitialBlockDownload(Params()))
         throw JSONRPCError(RPC_CLIENT_IN_INITIAL_DOWNLOAD, "Zcash is downloading blocks...");
 
+
+    MinerAddress minerAddress;
+    GetMainSignals().AddressForMining(minerAddress);
+
     static unsigned int nTransactionsUpdatedLast;
+    CMutableTransaction next_coinbase_tx;
+    bool have_next_coinbase_tx{false};
```

Comment:
Rather than a CMutableTransaction and a boolean value, why not use a std::optional here too?